upload/admin/controller/extension/shipping/mds.php

```diff
@@ -95,117 +68,46 @@ public function index() {
-            $data['shipping_mds_username'] = $this->request->post['shipping_mds_username'];
-        } else {
-            $data['shipping_mds_username'] = $this->config->get('shipping_mds_username');
-        }
-        if (isset($this->request->post['shipping_mds_password'])) {
-            $data['shipping_mds_password'] = $this->request->post['shipping_mds_password'];
-        } else {
-            $data['shipping_mds_password'] = $this->config->get('shipping_mds_password');
-        }
-        if (isset($this->request->post['shipping_mds_markup'])) {
-            $data['shipping_mds_markup'] = $this->request->post['shipping_mds_markup'];
-        } else {
-            $data['shipping_mds_markup'] = $this->config->get('shipping_mds_markup');
-        }
-        if (isset($this->request->post['shipping_mds_test'])) {
-            $data['shipping_mds_test'] = $this->request->post['shipping_mds_test'];
-        } else {
-            $data['shipping_mds_test'] = $this->config->get('shipping_mds_test');
-        }
-        if (isset($this->request->post['shipping_mds_insurance'])) {
-            $data['shipping_mds_insurance'] = $this->request->post['shipping_mds_insurance'];
-        } else {
-            $data['shipping_mds_insurance'] = $this->config->get('shipping_mds_insurance');
-        }
-        if (isset($this->request->post['shipping_mds_status'])) {
-            $data['shipping_mds_status'] = $this->request->post['shipping_mds_status'];
-        } else {
-            $data['shipping_mds_status'] = $this->config->get('shipping_mds_status');
-        }
-
-        if (isset($this->request->post['shipping_mds_is_demo'])) {
-            $data['shipping_mds_is_demo'] = $this->request->post['shipping_mds_is_demo'];
-        } else {
-            $data['shipping_mds_is_demo'] = $this->config->get('shipping_mds_is_demo');
-        }
-        if (isset($this->request->post['shipping_mds_tax_class_id'])) {
-            $data['shipping_mds_tax_class_id'] = $this->request->post['shipping_mds_tax_class_id'];
-        } else {
-            $data['shipping_mds_tax_class_id'] = $this->config->get('shipping_mds_tax_class_id');
+        $data['shipping_mds_status']                    = $this->getSettingValue('shipping_mds_status', 0);
+        $data['geo_zones']                              = $this->model_localisation_geo_zone->getGeoZones();
+        $data['shipping_mds_username']                  = $this->getSettingValue('shipping_mds_username', self::SANDBOX_USERNAME);
+        $data['shipping_mds_password']                  = $this->getSettingValue('shipping_mds_password', self::SANDBOX_PASSWORD);
+        $data['shipping_mds_is_demo']                  = $this->isSandBoxAccount();
+        $data['shipping_mds_test']                      = $this->getSettingValue('shipping_mds_test');
+        $data['shipping_mds_insurance']                 = $this->getSettingValue('shipping_mds_insurance');
```

Comment:
Just from a legal standpoint - stay away from calling it `insurance` only ever refer to it as `risk cover`